src/ServiceManagement/Compute/Commands.ServiceManagement/Extensions/Common/ExtensionRole.cs

```diff
@@ -63,12 +78,16 @@ public override string ToString()
 
         public string GetExtensionId(string extensionName, string slot, int index)
         {
-            var normalizedExtName = extensionName.Replace(".", string.Empty);
-            normalizedExtName = normalizedExtName.Replace(" ", string.Empty);
-            normalizedExtName = normalizedExtName.Replace("_", string.Empty);
+            var normalizedExtName = RemoveDisallowedCharacters(extensionName);
 
             var suffix = new StringBuilder();
             suffix.AppendFormat(ExtensionIdSuffixTemplate, normalizedExtName, slot, index);
+            if (suffix.Length > MaxSuffixLength)
+            {
+                int lenDiff = suffix.Length - MaxSuffixLength;
+                int startIndex = 1; // Suffix starts with '-'
+                suffix.Remove(startIndex + normalizedExtName.Length - lenDiff, lenDiff);
+            }
```

Comment:
Actually no... So the format of the extension ID is ```{role-name-part} + {suffix}```, where ```{suffix} = -{extension-name-part}-{slot}-Ext-{num}``` (all from legacy design request).

I want to limit the ```suffix``` length, but I can't simply cut it off by ```MaxSuffixLength```, and I have to truncate the ```{extension-name-part}```.